epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityItem.java

```diff
@@ -0,0 +1,169 @@
+package com.airbnb.epoxy;
+
+import android.graphics.Rect;
+import android.support.annotation.NonNull;
+import android.support.annotation.Px;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+class EpoxyVisibilityItem {
+
+  private final Rect localVisibleRect = new Rect();
+
+  private int adapterPosition = RecyclerView.NO_POSITION;
+
+  @Px
+  int size;
+  @Px
+  private int otherSize;
+
+  private boolean vertical;
+
+  private float percentVisibleSize = 0.f;
+  @Px
+  private int visibleSize;
+  @Px
+  private int viewportSize;
+
+  private boolean fullyVisible = false;
+  private boolean visible = false;
+  private boolean focusedVisible = false;
+
+  /**
+   * Update the visibility item according the current layout.
+   *
+   * @param view the current {@link com.airbnb.epoxy.EpoxyViewHolder}'s itemView
+   * @param parent the {@link android.support.v7.widget.RecyclerView}
+   * @param orientation the {@link android.support.v7.widget.LinearLayoutManager}'s orientation
+   *
+   * @return true if the view has been measured
+   */
+  boolean update(@NonNull View view, @NonNull RecyclerView parent, int orientation) {
+    view.getLocalVisibleRect(localVisibleRect);
+    vertical = orientation == LinearLayoutManager.VERTICAL;
```

Comment:
Do you support only LinearLayoutManager ?

Reply:
yes. btw a `GridLayoutManager` is also a `LinearLayoutManager `